### ISSUE TITLE
[JSX] Remove brackets around expressions of only elements

### ISSUE THREAD

**Issue by andrejewski** [prettier/prettier#6074]
**Prettier 1.17.0**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEAGvgBQCU+wAOlPvgE5wzr20A8AJjgG4B8wnHgHpeAXzZCufGqJAAaEBAAOuaAGdkoAIb16EAO4AFHQg0ot3CDg7yQAI3pawAayYBlJU5xQA5shj06HAKABYwALYANgDqITjwap5gcG6m8TzxAJ7I4GoaCt5qcPQwho4+4VrIqFqRRQoAVmoAHgBCji7uWuFwADLecNW19SBNzW7ePpFwAIroEPBDdcEgnvRF9Dl2WnaZkdC2SvTeMNHWMCHIABwADApHEEXRjko5R3Ab3IMKAI7z8GVlGYQFo1ABaKBwOAcaG2Rh-HCMMpaCpVJA1ZYKIrhHD+QIrNSTaZzBaDdHDFYwHZnDgXZAAJgUAS0OEikwAwhBwpUclBoN8QOgigAVHZmDFFUSiIA)
```sh
--parser babylon
```

**Input:**
```jsx
function X() {
  return <div>{<div />}</div>;
}
```

**Output:**
```jsx
function X() {
  return <div>{<div />}</div>;
}

```

**Expected behavior:**
```jsx
function X() {
  return <div><div /></div>;
}

```
---

**Reasoning**

In these cases where there is only an element within braces, the braces provide no value and are most often an artifact of removing an `isX && ...` or other condition. It would be great for Prettier to sweep up these unnecessary brackets.

**Comment by j-f1** [comment]
This is a good idea, but I’m not sure if it’s OK to change this here. We try not to alter the AST where possible, and changing this definitely changes the AST, so :shrug:

**Comment by rattrayalex** [comment]
Hmm, we already modify the AST in a very similar way for whitespace for JSX. eg; `<div> <span /></div>` can become `<div>{' '}<span /></div>` and vice versa. 

That being said, the space has a compelling use case, where unwrapping `{` is, afaik, purely aesthetic – and thus perhaps is best served by an eslint autofix.

**Comment by j-f1** [comment]
Also, we preserve this decidedly strange formatting:

**Prettier 1.16.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEAGvgBQCU+wAOlPvgE5wzr20A8AJjgG76oQQAvJx4B6AHz4JNAL4gANCAgAHXNADOyUAEN69CAHcACroSaU27hBwcFIAEb1tYANZMAysuc4oAc2Qw9OhwigAWMAC2ADYA6qE48OpeYHDuZgk8CQCeyODqmoo+6nD0MEZOvhHayKjaUcWKAFbqAB4AQk6uHtoRcAAyPnA1dQ0gzS3uPr5RcACK6BDww-UhIF70xfS59tr2WVHQdsr0PjAxNjChyAAcAAyKxxDFMU7Kucdwm9xDiowAjugcIxytpKtUkLUVopihEcAEgqt1FMZvNFkMISNVjBducOJdkAAmRSBbQ4KJTADCEAiVVyUGgPxA6GKABVduZIaMAWjyipzCBtOoALRQOBwDjikAyGRAA)
```sh
--parser babylon
```

**Input:**
```jsx
function X () {
  return <div foo=<div/> />
}
```

**Output:**
```jsx
function X() {
  return <div foo=<div /> />;
}

```

**Comment by duailibe** [comment]
Yeah, I don't think that's a compelling to add this. We already have people complaining we change `<div></div>` to `<div />`, so they have to change it back to add a child later.

**Comment by hawkrives** [comment]
Personally, and I don' tknow if this is a compelling argument or not, but I view braces-in-jsx-around-jsx as being very similar to how Prettier removes parentheses from expressions. IE, they're extraneous, so there's no reason for them to be there.

I understand that the parentheses aren't actually in the AST, while these braces are, but I would very much like to have Prettier remove unneeded braces from JSX bodies.

**Comment by duailibe** [comment]
@hawkrives I kind of agree with you.

The rationale is we've always been on the safe side of things, doing as little transformation as possible to ensure we don't break stuff. To remove these `{}`, we'd have to keep track of when/where it's safe and carrying the risk of erroring sometimes.

(Somewhat unrelated, we have the issue of assuming we can always format what's inside a ` html`` ` or ` graphql`` ` and others; but we found out later that it's not always true)
